blacklite-core/src/main/java/com/tersesystems/blacklite/AsyncEntryWriter.java

```diff
@@ -89,19 +92,49 @@ public AsyncEntryWriter(
                 try {
                   entryStore.insert(e.epochSecond, e.nanos, e.level, e.content);
 
-                  // Always check to see if we should be committing here...
-                  if (inserts.incrementAndGet() > batchInsertSize) {
-                    entryStore.executeBatch();
-                    inserts.set(0);
+                  // Always flush on batch insert size, even if we've never been idle.
+                  if (inserts.incrementAndGet() >= batchInsertSize) {
+                    commit(inserts);
                   }
+                  archive(lastRun);
                 } catch (SQLException ex) {
                   statusReporter.addError(ex.getMessage(), ex);
                 }
               };
-          queue.drain(consumer, waitStrategy, () -> (acceptingWrites() || !queue.isEmpty()));
+          queue.drain(consumer, onIdle, () -> (acceptingWrites() || !queue.isEmpty()));
+          if (tracing) {
+            statusReporter.addInfo("AsyncEntryWriter: queue no longer accepting writes");
+          }
         });
   }
 
+  private void commit(AtomicLong inserts) throws SQLException {
```

Comment:
break out commits to a different method, make it clearer that inserts happen up to batchInsertSize and then `executeBatch/commit` always happens